d2-schemas/src/main/pegasus/com/linkedin/d2/D2RingProperties.pdl

```diff
@@ -0,0 +1,92 @@
+namespace com.linkedin.d2
+
+/**
+ * D2 hash ring properties used for load balancing.
+ */
+record D2RingProperties {
+
+  /**
+   * How fast should load balancer ramp up node that has been 100% degraded and is being recovered. Defaults to 1.0.
+   */
+  ringRampFactor: optional double
+
+  /**
+   * The number of points per weight given for a particular node in a hash ring. Defaults to 100.
+   */
+  pointsPerWeight: optional int
+
+  /**
+   * What hashing method used to hash incoming request. Used to determine which node should serve the request. Choices are Random, UriRegex
+   */
+  @symbolDocs = {
+    "URI_REGEX" : "Extract a key from URI and use it to hash",
+    "RANDOM" : "The default fall back value"
+  }
+  hashMethod: optional enum hashMethodEnum {
+    RANDOM
+    URI_REGEX
+  }
+
+  /**
+   * Configuration used to supplement the hash method.
+   */
+  hashConfig: optional record hashConfigType {
+
+    /**
+     * List of Regex to match against the URI of incoming request and compute hash value. The hash value is computed based on the contents of the first capture group of the first expression that matches the request URI
+     */
+    uriRegexes: optional array[string]
+
+    /**
+     * Optional config value. if true, fail if no regex matches, otherwise fall back to random.
+     */
+    failOnNoMatch: boolean = false
+
+    /**
+     *  Optional config value. if false, don't warn on falling back to random if the uri doesn't match the regex
+     */
+    warnOnNoMatch: boolean = true
+  }
+
+  /**
+   * The highest ratio of unused entries over the total entries of the Ring points that d2 maintains.
```

Comment:
Discussed offline about defaults, I want to not include defaults in pdl to avoid having an extra layer of "defaults". Currently the priority is:
1. user-configured cfg2
2. global cfg2
3. code
4. pdl

I'd rather just get rid of all pdl defaults since they are commonly overridden anyway and will be more misleading than helpful.